### PR TITLE
changing plugin manager with redefined plugin structure

### PR DESCRIPTION
**Pull request by JaydipGabani**

### CODE REVIEW

File: cmd/plugin-manager/add/add.go [hunk 1]
```diff
@@ -135,22 +137,30 @@ func (o *Options) run(args []string) error {
...
-			case len(manifest) > 1:
+			case len(pluginsMap[args[0]]) > 1:
 				// if there are multiple version of the plugins are available then look for the latest or mentioned version and if not found fail and ask user to input a version using --version flag
-				for _, value := range manifest {
+				if installVersion == "" {
+					availableVersions := []string{}
+					for _, value := range pluginsMap[args[0]] {
+						availableVersions = append(availableVersions, string(value.Version))
+					}
+					semver.Sort(availableVersions)
```
Comment (djzager) [inline cmd/plugin-manager/add/add.go:160]: :tada:

File: internal/plugin/plugin_index_types.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package plugin
+
+type PluginLocation struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+type PluginIndex struct {
```
[the remark this answers is not in the document]
Comment (djzager) [reply internal/plugin/plugin_index_types.go:8]: I see now. I think all of these can go into plugin_types.

File: internal/plugin/plugin_manager_helper.go [hunk 3]
```diff
@@ -11,37 +11,18 @@ import (
...
 	"github.com/ghodss/yaml"
-	"github.com/konveyor/crane-lib/transform"
 	"github.com/sirupsen/logrus"
 )
 
 const (
 	DEFAULT_REPO     = "default"
 	DEFAULT_REPO_URL = "DEFAULT_REPO_URL"
-	DEFAULT_URL      = "https://raw.githubusercontent.com/konveyor/crane-plugins/crane-cli/index.yml"
+	DEFAULT_URL      = "https://raw.githubusercontent.com/JaydipGabani/crane-plugins/main/index.yaml"
```
[the remark this answers is not in the document]
Comment (JaydipGabani) [reply internal/plugin/plugin_manager_helper.go]: ack

**Review by djzager — APPROVED** [review]
/lgtm

**Pull request merged**